vendor_data.py

```diff
@@ -1,241 +1,375 @@
+
+    def SetStatus(self, PrefVend):
+        """Method
+
+        Args:
+            PrefVend (bool): new value for Preferred, has an influence on status
+        """
+        if PrefVend != None:
+            if type(PrefVend) != bool:
+                raise TypeError("Preferred should either be True, False or None.")
+        if PrefVend != None:
+            if PrefVend == True:
+                self.Preferred = True
+                self.Status = "Preferred"
+            elif PrefVend == False:
+                self.Preferred = False
+                self.Status = "Back-up"
         else:
-            self.CatTool = tool
-    
+            self.Preferred = None
+            self.Status = "Potential"             
    
-    def to_excel (self):
-        data = {
+    def ToExcel(self): #method to write data to an excel file
+        Data = { #dump arguments provided in a dictionary
             "Target Language": [self.TargLang],
             "Vendor": [self.VendorName],
             "E-mail": [self.VendorMail],
             "CAT Tool": [self.CatTool],
             "Word Rate": [self.WordRate],
             "Status": [self.Status]
         }
-        filename = "_".join([str(self.ProjectName), str(self.SourceLang)]) + ".xlsx"
-        df = pd.DataFrame(data)
-        if os.path.exists(filename) != True:
-            df.to_excel(filename, index=False, sheet_name="VendorData")
-        else:
-            with pd.ExcelWriter(filename, mode="a", engine="openpyxl", if_sheet_exists="overlay") as writer:
+        FileName = "_".join([str(self.ProjectName), str(self.SourceLang)]) + ".xlsx" #the name of the excel file, based on arguments `ProjectName` and `SourceLang`
+        df = pd.DataFrame(Data) #make a panda dataframe out of the dictionary
+        if not os.path.exists(FileName): #check if the file exists, if not a new file is written
+            df.to_excel(FileName, index=False, sheet_name="VendorData")
+        else: #if file exists, the data is appended to the existing file
+            with pd.ExcelWriter(FileName, mode="a", engine="openpyxl", if_sheet_exists="overlay") as writer:
                 df.to_excel(writer, sheet_name = "VendorData", startrow=writer.sheets["VendorData"].max_row, header = None, index=False)
+    def ReadExcel(self): #method to read the file with filename {ProjectName_SourceLang} into a dictionary
+        FileName = "_".join([str(self.ProjectName), str(self.SourceLang)]) + ".xlsx"
+        if os.path.exists(FileName):
+            ExcelRecords = pd.read_excel(FileName) #read filename
+            ExcelRecordsDf = ExcelRecords.loc[:, ~ExcelRecords.columns.str.contains('^Unnamed')] #remove any unnamed columns
+            VendorDict = ExcelRecordsDf.to_dict() #turn dataframe into a dictionary
+            return VendorDict #print the Dictionary
+        else:
+            raise ValueError(f"There is no file for {self.ProjectName} in {self.SourceLang}")
+    
+    def ModExcel(self, Key, Index, NewValue):

```

Comment:
Good job with this method.
BUT. This means that if a user has changed two or three values in an entry, you will open and close the Excel file two or three times, one for each value that you want to change.
If you think that users will not change more than one thing at a time, it's ok, but it might make more sense to collect all the new values and update them in one opening-closing of the file.